### PR TITLE
feat: create initial setup for e2e + gh automation

### PR DESCRIPTION
**Pull request by MCarlomagno**
# Summary

Created an initial e2e setup for testing the package against the server. `e2e` folder is a standalone project, just like `examples`, that uses current workspace sdk version to make API calls.

To run e2e tests:

```sh
pnpm test:e2e
```

The initial file structure is:

```sh
/e2e/src ___ actions.test.ts
        |___ account.test.ts
        |___ ...
```

Also created initial gh automation that runs when a PR is merged.

## Running locally

Go to `e2e` folder and create `.env` file, copy `.env.example` entries and set staging credentials. Then run `pnpm install` + `pnpm test:e2e`. We are going to use `DEFAULT_TENANT` credentials for automation.

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - name: Checkout
+      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.5.4
+
+    - name: Prepare pre-requisites
+      uses: ./.github/actions/prepare
+
+    - name: Build
+      run: |
+        pnpm nx-build-skip-cache
+
```
Comment (MCarlomagno) [inline .github/workflows/e2e.yml:28]: TODO: add step to include env variables

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# Declare default permissions as read only.
+permissions: read-all
+
+env:
+  API_KEY: ${{ secrets.API_KEY }}
+  API_SECRET: ${{ secrets.API_SECRET }}
+  DEFENDER_API_URL: ${{ secrets.DEFENDER_API_URL }}
+  DEFENDER_POOL_ID: ${{ secrets.DEFENDER_POOL_ID }}
+  DEFENDER_POOL_CLIENT_ID: ${{ secrets.DEFENDER_POOL_CLIENT_ID }}
```
Comment (MCarlomagno) [inline .github/workflows/e2e.yml:19]: @tirumerla I think I don't have permission to create secrets in this repo

File: e2e/src/actions.test.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+describe('actions', () => {
+  const actionClient = new Defender({
+    apiKey: process.env.API_KEY,
+    apiSecret: process.env.API_SECRET,
+  }).action;
+
+  it('should list actions', async () => {
+    const result = await actionClient.list();
+    expect(result.items).toBeDefined();
+  });
```
Comment (MCarlomagno) [inline e2e/src/actions.test.ts:16]: Example e2e test

**Review by dylankilkenny — APPROVED** [review]

**Review by shahnami — APPROVED** [review]

**Review by CoveMB — APPROVED** [review]
Good work ⚙️

**Review by zeljkoX — APPROVED** [review]
LGTM

Thanks!

**Pull request merged**